### PR TITLE
Feature/productionise multi selfie cx 3922

### PR DESCRIPTION
**Pull request by stefaniacardenas**
# Problem
Currently, the `useMultipleSelfieCapture` configuration for `face` step is not public. This PR will make the feature production ready
# Solution
Feature description from README
```
- useMultipleSelfieCapture (boolean - default: `false`)
By enabling this configuration, the SDK will take a snapshot of the applicant, along side the selfie photo. The snapshot will be analysed by our ML algorithm and compared against the selfie photo in order to reduce fraud attempts.
```
This PR involves:
- adding documentation on README
- add automated testing
- send analytics to woopra

Bonus: as part of this task we added automated tests for camera stream 🎉 

## Checklist
_put `n/a` if item is not relevant to PR changes_

- [x] Have the CHANGELOG, README, MIGRATION, RELEASE_GUIDELINES docs been updated?
- [ ] Have new automated tests been implemented?
- [ ] Have new manual tests been written down?
- [x] Have tests passed locally?
- [ ] Have any new strings been translated?

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

File: test/pageobjects/DocumentUploadConfirmation.js [hunk 1]
```diff
@@ -79,9 +81,15 @@ class DocumentUploadConfirmation extends Base {
     verifyElementCopy(this.errorInstruction, documentUploadConfirmationErrorStrings.glare_detected.instruction)
   }
 
+  // TODO: this function is never called!
```
Comment (bartzet) [inline test/pageobjects/DocumentUploadConfirmation.js]: Let's remove it.

Comment (stefaniacardenas) [reply test/pageobjects/DocumentUploadConfirmation.js]: No, I started using it as part of this PR. I need to remove the comment and make sure I refactor the other tests to use this function :)

**Comment by rfreitas** [comment]
Travis automatic deployment: https://760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

File: MANUAL_REGRESSION.md [hunk 2]
```diff
@@ -492,23 +492,6 @@ Given user opened the link with `?uploadFallback=false` flag
     - "Check readability" text and back arrow retain the colour
     - Back navigation in the browser doesn't cause any other UI changes in the SDK
 
-## Internal - functional
```
Comment (bartzet) [inline MANUAL_REGRESSION.md]: 🔥

File: test/pageobjects/DocumentUploadConfirmation.js [hunk 3]
```diff
@@ -11,6 +11,8 @@ class DocumentUploadConfirmation extends Base {
   get errorTitleIcon() { return this.$('.onfido-sdk-ui-Error-title-icon-error')}
   get warningTitleIcon() { return this.$('.onfido-sdk-ui-Error-title-icon-warning')}
   get errorInstruction() { return this.$('.onfido-sdk-ui-Error-instruction-text')}
+  get shutterButton() { return this.$('.onfido-sdk-ui-Photo-btn')}
+
```
Comment (bartzet) [inline test/pageobjects/DocumentUploadConfirmation.js]: Extra line added.

File: test/main.js [hunk 4]
```diff
@@ -34,18 +34,28 @@ const browserstackLocalDefault = {
 const currentDate = Date.now().toString();
 const random = () => Math.random().toString(36).substring(7)
 
-const createDriver = ({name,localIdentifier}) => browser =>
+const chromeCapabilities = Capabilities.chrome()
+const chromeOptions = {
+  'args': ['--use-fake-device-for-media-stream','--use-fake-ui-for-media-stream', `--use-file-for-fake-video-capture=${__dirname}/resources/test-stream.y4m`]
```
Comment (bartzet) [inline test/main.js]: This is great 💯

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://760-pr-onfido-sdk-ui-onfido.surge.sh

File: test/specs/happy.js [hunk 5]
```diff
@@ -177,6 +176,24 @@ describe('Happy Paths', options, ({driver, pageObjects}) => {
...
+
+      it('should take multiple selfies using the camera stream', async () => {
+        goToPassportUploadScreen(`?language=${lang}&async=false&useMultipleSelfieCapture=true`)
+        uploadFileAndClickConfirmButton('passport.jpg')
+        documentUploadConfirmation.takeSelfie()
+        documentUploadConfirmation.confirmBtn.click()
+        verificationComplete.verifyUIElements(verificationCompleteCopy)
+        verificationComplete.checkBackArrowIsNotDisplayed()
+      })
+
```
Comment (kopijunkie) [inline test/specs/happy.js]: How are we confirming if multiple selfies were taken for this test?
Code wise looks pretty similar.

Comment (stefaniacardenas) [reply test/specs/happy.js]: The second test uses the `?useMultipleSelfieCapture=true` query string which uses the option to send 2 requests for selfie. The fact that the user can get to the end of the flow means that both requests succeed. If one of the two requests fails the user will see an error. This tests the happy path. In terms of UX/UI nothing changes between standard capture and multiple frame capture.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://760-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-760-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**